### PR TITLE
Support stellar-core protocol 12.

### PR DESCRIPTION
**Pull request by abuiles**
Add support for `path_payment_strict_send` and `path_payment_strict_receive`

### Testing 

You cant test this PR installing directly from this branch and then running scc.

#### Install 

```
git clone git@github.com:stellar/stellar_core_commander.git
git checkout core12-support
gem build stellar_core_commander.gemspec
gem install stellar_core_commander-0.0.13.gem
```

#### Run scc

Copy the following code into an file called `path_payment.rb`

```
account :usd_gateway
account :eur_gateway
account :scott
account :bartek
account :andrew

create_account :usd_gateway, :master
create_account :eur_gateway, :master
create_account :scott, :master
create_account :bartek, :master
create_account :andrew, :master

close_ledger

trust :scott,  :usd_gateway, "USD"
trust :bartek, :eur_gateway, "EUR"
trust :andrew, :usd_gateway, "USD"
trust :andrew, :eur_gateway, "EUR"

close_ledger

payment :usd_gateway, :scott,  ["USD", :usd_gateway, 1000]
payment :usd_gateway, :andrew, ["USD", :usd_gateway, 200]
payment :eur_gateway, :andrew, ["EUR", :eur_gateway, 200]
payment :eur_gateway, :bartek, ["EUR", :eur_gateway, 1000]

close_ledger

offer :andrew, {buy:["USD", :usd_gateway], with:["EUR", :eur_gateway]}, 200, 1.0

close_ledger

path_payment_strict_send :scott, :bartek, ["EUR", :eur_gateway, 10], with: ["USD", :usd_gateway, 10], path: []
path_payment_strict_receive :scott, :bartek, ["EUR", :eur_gateway, 10], with: ["USD", :usd_gateway, 10], path: []

close_ledger
```

And then run `scc -r path_payment.rb --allow-failed-transactions --dump-root-db > path.sql`

### CODE REVIEW

File: lib/stellar_core_commander/transaction_builder.rb [hunk 1]
```diff
@@ -80,14 +80,54 @@ def payment(from, to, amount, options={})
       tx =  if options[:with]
               attrs[:with] = normalize_amount(options[:with])
               attrs[:path] = options[:path].map{|p| make_asset p}
-              Stellar::Transaction.path_payment(attrs)
+              Stellar::Transaction.path_payment_strict_receive(attrs)
```
Comment (ire-and-curses) [inline lib/stellar_core_commander/transaction_builder.rb]: Does `path_payment_strict_send` not also need to be included here?

Comment (ire-and-curses) [reply lib/stellar_core_commander/transaction_builder.rb]: Or is handled in a different way?

Comment (abuiles) [reply lib/stellar_core_commander/transaction_builder.rb]: @ire-and-curses it's handled in a different way, now the commander has `path_payment_strict_send` -- this keeps backwards compatibility with the `payment`  API, which could let you do a normal payment or a pathed payment.

Comment (abuiles) [reply lib/stellar_core_commander/transaction_builder.rb]: See https://github.com/stellar/stellar_core_commander/pull/138/files#diff-8a33cbcbfad87a49cfb1de064b789286R33

Comment (ire-and-curses) [reply lib/stellar_core_commander/transaction_builder.rb]: Makes sense thanks :+1:

**Pull request merged**